leader/src/jerigon.rs

```diff
@@ -1,61 +1,64 @@
-use std::{
-    fs::{create_dir_all, File},
-    io::Write,
-    path::PathBuf,
-};
+use std::path::PathBuf;
 
 use alloy::providers::RootProvider;
 use anyhow::Result;
 use common::block_interval::BlockInterval;
+use common::fs::generate_block_proof_file_name;
 use paladin::runtime::Runtime;
 use proof_gen::proof_types::GeneratedBlockProof;
+use tracing::{error, info};
+
+#[derive(Debug, Default)]
+pub(crate) struct JerigonParams {
+    pub checkpoint_block_number: u64,
+    pub previous_proof: Option<GeneratedBlockProof>,
+    pub proof_output_dir: Option<PathBuf>,
+    pub save_inputs_on_error: bool,
+    pub keep_intermediate_proofs: bool,
+}
```

Comment:
minor: may deserve to live outside of `jerigon.rs` as this would also be used with the `native` tracer (whichever gets merged first into `develop`), and be renamed to `ProofParams` or something else more neutral.